### PR TITLE
Implement Town::update at 0x0049742F

### PR DESCRIPTION
**Pull request by petergaal**

### CODE REVIEW

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
Apologies for the delay in testing this.

Unfortunately, replay tests fail on the [Randomland by matsku84 - 2050.SV5.zip](https://github.com/OpenLoco/OpenLoco/files/7527061/Randomland.by.matsku84.-.2050.SV5.zip) save. Leave the game running for an in-game month to see a divergence in the autosave, compared to doing the same thing on master.

Local comparison for the second autosave generated using Beyond Compare:
```
Hex Compare
Produced: 12/11/21 12:09:33 pm 

Left file: /home/aaron/Repositories/OpenLoco/OpenLoco/build/config/save/autosave/matsku-2050-master-compressed-post-1101/autosave_2021-08-12_13-46-41.SV5
Right file: /home/aaron/Repositories/OpenLoco/OpenLoco/build/config/save/autosave/autosave_2021-11-12_12-05-47.SV5
2248645 same byte(s)
5305 left orphan byte(s)
5306 right orphan byte(s)
8686 difference byte(s)
```

File: src/OpenLoco/Town.cpp [hunk 1]
```diff
@@ -13,12 +13,29 @@ namespace OpenLoco
...
     void Town::update()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        call(0x0049742F, regs);
+        recalculateSize();
+
+        static const std::array<uint8_t, 12> buildSpeedToGrowthPerTick = { 0, 1, 3, 5, 7, 9, 12, 16, 22, 0, 0, 0 };
+        auto buildSpeed = buildSpeedToGrowthPerTick[this->build_speed];
+        if (buildSpeed == 0 || (buildSpeed == 1 && (gPrng().randNext() & 7) == 0))
```
Comment (duncanspumpkin) [inline src/OpenLoco/Town.cpp:28]: this should be `buildSpeed == 0 || (buildSpeed == 1 && (gPrng().randNext() & 7))`

**Comment by duncanspumpkin** [comment]
Can you recheck this @AaronVanGeffen

**Comment by duncanspumpkin** [comment]
Confirmed no divergence against master.

**Pull request merged**